### PR TITLE
Added argument to the mmCIF parser to predefine the use of label_seq_id

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #2259 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This should address the issue, I tested it on Python3.6; it would be good is someone else review and test it.

### CODE REVIEW

**Comment by jgreener64** [comment]
This more or less looks good to me, I have made the odd comment on the code.

Could you add a test case, ideally re-using an existing mmCIF file in the test suite?

Could you also add a sentence to the docs on this, perhaps at the end of section 11.1.1?

File: Bio/PDB/MMCIFParser.py [hunk 1]
```diff
@@ -141,7 +143,12 @@ def _build_structure(self, structure_id):
             altloc = alt_list[i]
             if altloc in _unassigned:
                 altloc = " "
-            int_resseq = int(seq_id_list[i])
+            try:
+                int_resseq = int(seq_id_list[i])
+            except ValueError:
+                raise PDBConstructionException(
+                    "try parser.get_structure('structure_id', 'file_name', use_label_seq_id=True)"
```
Comment (jgreener64) [inline Bio/PDB/MMCIFParser.py]: Perhaps clarify this error, something like "conversion of sequence ID to integer failed, consider using...".

**Comment by svalqui** [comment]
I'm afraid this fails, on files that have HETATM information, the field _atom_site.label_seq_id is sometimes a dot ".".

from Tests/PDB/2XHE.cif

ATOM   6267 C CG2 . VAL B 2 261 ? 17.741  -74.246 1.436   1.00 209.01 ? 261  VAL B CG2 1
HETATM 6268 O O   . HOH C 3 .   ? -17.130 -57.437 4.024   1.00 70.68  ? 2001 HOH A O   1

I do not seem to find another way than changing
https://github.com/biopython/biopython/blob/dbdfd1977d48c503c64a6669f8a8785e4f18346f/Bio/PDB/MMCIFParser.py#L144

which would expand to StructureBuilder.py, and that further expands to other files.
Is a significant change to the module.

I'll think about this for a couple of weeks to see if I can afford the time.

**Comment by peterjc** [comment]
Closed as per https://github.com/biopython/biopython/issues/2259#issuecomment-546780021 which describes the larger body of work needed.

**Pull request closed without merging**